### PR TITLE
[Agent] recover l7 fail flow

### PR DESCRIPTION
**Pull request by qq906907952**
### This PR is for:
- Agent

#### Affected branches
- main
- v6.3
#### Checklist
- [ ] Added unit test to verify the fix.
- [ ] Verified eBPF program runs successfully on linux 4.14.x.
- [ ] Verified eBPF program runs successfully on linux 4.19.x.
- [ ] Verified eBPF program runs successfully on linux 5.2.x.

### CODE REVIEW

File: agent/src/flow_generator/app_table.rs [hunk 1]
```diff
@@ -167,17 +164,14 @@ impl AppTable {
             {
                 return None;
             } else {
-                return Some((
-                    v.l7_protocol_enum.clone(),
-                    self.l7_protocol_inference_max_fail_count,
-                ));
+                return Some((v.l7_protocol_enum.clone(), v.unknown_count, v.last));
```
Comment (qq906907952) [inline agent/src/flow_generator/app_table.rs:167]: @yuanchaoa  这里你是不是写错了 ，不应该返回self.l7_protocol_inference_max_fail_count,

File: agent/src/flow_generator/app_table.rs [hunk 2]
```diff
@@ -268,14 +272,19 @@ impl AppTable {
         let key = AppTable4Key { ip, epc, port, pid };
         let value = self.ipv4.get_mut(&key);
         if let Some(value) = value {
-            value.last = time_in_sec;
             if l7_protocol_enum.get_l7_protocol() == L7Protocol::Unknown {
+                if value.last + self.l7_protocol_inference_ttl < time_in_sec {
+                    value.unknown_count = 0;
```
Comment (qq906907952) [inline agent/src/flow_generator/app_table.rs:277]: @yuanchaoa 这里对fail count的上报有没有影响

File: agent/src/flow_generator/app_table.rs (hunk 1 above)
Comment (yuanchaoa) [reply agent/src/flow_generator/app_table.rs:167]: 对 是有问题

File: agent/src/flow_generator/perf/mod.rs [hunk 3]
```diff
@@ -201,11 +201,28 @@ pub struct FlowLog {
...
 }
 
 impl FlowLog {
     const PROTOCOL_CHECK_LIMIT: usize = 5;
 
+    // if flow parse fail exceed l7_protocol_inference_max_fail_count and time exceed l7_protocol_inference_ttl,
+    // recover the flow check and parse
+    fn check_fail_recover(&mut self) {
+        if self.is_skip {
+            let now = SystemTime::now();
```
Comment (yuanchaoa) [inline agent/src/flow_generator/perf/mod.rs:217]: 获取的时间戳需要用ntp_diff校准

**Review by yuanchaoa — APPROVED** [review]

**Pull request merged**